### DISCUSSION TITLE
Remember locale - adjust all links to append the current locale to url

### DISCUSSION THREAD

**Discussion by kombuchapunk** [statamic/cms#4404] [Troubleshooting]
Hi! I'm building a multisite with a language dropdown done this way:

```
<select class="form-select language" onchange="javascript:changeLanguage(this)" aria-label="English">
	{{ locales }}
		{{ if is_current }}
			<option selected value="{{ permalink }}">{{ locale:name }}</option>
		{{ else }}
			<option value="{{ permalink }}">{{ locale:name }}</option>
		{{ /if }}
	{{ /locales }}
</select>
<script type="text/javascript">
      // CHANGE LANGUAGE
      function changeLanguage(el) {
	      window.location = el.value;
      }
</script>
```

where everything seems to work fine but every time I change the locale and then click on any menu item, I automatically get to the default english version. I don't want to rebuild all links for all 100 pages for each language and want the links in pages and in top navbar to adjust based on the currently chosen locale/language since the slug structure won't change in the site.
It looks like all of the pages and stuff are linked using ids and each localized page version has its own id so this is a bit confusing.

I can provide any additional information if needed.

Also, what's a good way to check the user's default browser locale and initially set it as a main one on first site's launch?

Thank you!

**Comment by netwaretcs** [comment]
I've the same problem. Have you found a solution? @MishaPetrov
